### ISSUE TITLE
Why is CLIPImageProcessor not in general init?

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/transformers#20122]
### System Info

- `transformers` version: 4.25.0.dev0
- Platform: Linux-5.18.10-76051810-generic-x86_64-with-glibc2.34
- Python version: 3.9.7
- Huggingface_hub version: 0.11.0.dev0
- PyTorch version (GPU?): 1.11.0+cpu (False)
- Tensorflow version (GPU?): 2.9.1 (False)
- Flax version (CPU?/GPU?/TPU?): 0.6.0 (cpu)
- Jax version: 0.3.16
- JaxLib version: 0.3.15
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@sgugger maybe

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In diffusers we load transformers classes according to a model index, which *e.g.* looks as follows:

```bash
{
  "_class_name": "StableDiffusionPipeline",
  "_diffusers_version": "0.7.0.dev0",
  "feature_extractor": [
    "transformers",
    "CLIPFeatureExtractor"
  ],
  "scheduler": [
    "diffusers",
    "PNDMScheduler"
  ],
  "text_encoder": [
    "transformers",
    "CLIPTextModel"
  ],
  "tokenizer": [
    "transformers",
    "CLIPTokenizer"
  ],
  "unet": [
    "diffusers",
    "UNet2DConditionModel"
  ],
  "vae": [
    "diffusers",
    "AutoencoderKL"
  ]
}
```

The important part is:

```
  "feature_extractor": [
    "transformers",
    "CLIPFeatureExtractor"
  ],
```

Now what is happening then is that we load a component that we call `"feature_extractor"` from `"transformers"` and the `"CLIPFeatureExtractor"` class. Then when saving the model we save it with `type(feature_extractor)` which is now though `CLIPImageProcessor` and then we want to load it again from `transformers`, but we cannot import it from transformers.

E.g. `from transformers import CLIPImageProcessor` doesn't work.

Could we add `CLIPImageProcessor` to the init? 

### Expected behavior

I think we should put `CLIPImageProcessor` in the init no?

**Comment by sgugger** [comment]
Will be fixed by #20111

**Comment by ZoeyyHz** [comment]
I have the same problem... God...When will the CLIPImageProcessor be set up in general init?

**Comment by amyeroberts** [comment]
@ZoeyyHz, which version of transformers are you using? I'm able to run the following without issue:

```
from transformers import CLIPImageProcessor
```

**Comment by BigTail375** [comment]
Hi, there. I have the same problem too.
which version of transformers do I have to use?

**Comment by amyeroberts** [comment]
@BigTail375 The CLIPImageProcessor has been available to import from the public init from v4.25.1
